lib/radiator/directory/editor/permission.ex

```diff
@@ -51,43 +51,49 @@ defmodule Radiator.Directory.Editor.Permission do
   def has_permission(user, subject, permission) do
     case PermissionType.compare(get_permission(user, subject), permission) do
       :lt ->
-        case parent(subject) do
-          nil ->
-            false
+        Enum.any?(parents(subject), fn parent ->
+          case parent do
+            nil ->
+              false
 
-          parent ->
-            has_permission(user, parent, permission)
-        end
+            parent ->
+              has_permission(user, parent, permission)
+          end
+        end)
 
       # greater or equal is fine
       _ ->
         true
     end
   end
 
-  # fixme: audio might have multiple parents:
-  #        - either one of possibly many episodes
-  #        - or network
-  defp parent(subject = %Audio{}) do
-    subject
-    |> Ecto.assoc(:episodes)
-    |> Repo.one!()
+  defp parents(subject = %Audio{}) do
+    network =
+      subject
+      |> Ecto.assoc(:network)
+      |> Repo.one()
+
+    episodes = subject |> Ecto.assoc(:episodes) |> Repo.all()
+
+    [network | episodes]
```

Comment:
hmmm, so in your implementation this can return nil in the list. I don't think this is a good contract. Would only add network if not nil.